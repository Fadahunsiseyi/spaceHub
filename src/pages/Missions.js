import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchMission } from '../redux/missions/missionReducer';
import Mission from '../components/Mission';

let initial = true;

const Missions = () => {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missionReducer);

  useEffect(() => {
    if (initial) {
      dispatch(fetchMission());
      initial = false;
    }
  }, [dispatch]);

  return (
    <div className="mission-container-list">
      <table>
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th> </th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => (
            <Mission
              key={mission.mission_id}
              id={mission.mission_id}
              name={mission.mission_name}
              description={mission.description}
              join={mission.join}
            />
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default Missions;
