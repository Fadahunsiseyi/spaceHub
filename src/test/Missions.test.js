import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import React from 'react';
import Missions from '../pages/Missions';
import store from '../redux/configureStore';

describe('Pages renders correctly', () => {
  it('Missions page renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>,
      )
      .toJSON();
    render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    const element = document.querySelector('.mission-container-list');
    expect(element).toBeTruthy();
    expect(tree).toMatchSnapshot();
  });
});
