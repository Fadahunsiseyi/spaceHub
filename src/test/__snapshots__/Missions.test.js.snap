// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages renders correctly Missions page renders correctly 1`] = `
<div
  className="mission-container-list"
>
  <table>
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th>
           
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
